packages/file-explorer/src/FileList.tsx

```diff
@@ -441,9 +444,23 @@ export const FileList = (props: FileListProps): JSX.Element => {
         setSelectedRanges(newSelectedRanges);
         const selectedItems = getItems(newSelectedRanges);
         onSelectionChange(selectedItems);
+        if (
+          selectedItems.length === 1 &&
+          selectedItems[0].type === 'directory'
+        ) {
+          pathGetter?.(selectedItems[0]);
+        } else {
+          const root: FileStorageItem = {
+            type: 'directory',
+            id: '/',
+            filename: '/',
+            basename: '',
+          };
+          pathGetter?.(root);
+        }
```

Comment:
Instead of making a new prop `pathGetter` and passing that down, you should instead pipe `onSelectionChange` through so it gets back up to `FileExplorerPanel`.
That'll mean adding the prop to `FileListContainer`, and then in `FileListContainer.handleSelectionChange`, calling `onSelectionChange` after it calls `setSelectedItems`. 

Then all this logic goes into `FileExplorerPanel`, where it decides whether to update the path if there's only one selected item and it's a dir.